apps/src/dance/Dance.js

```diff
@@ -560,6 +561,8 @@ Dance.prototype.updateSongMetadata = function (id) {
  */
 Dance.prototype.onHandleEvents = function (currentFrameEvents) {
   this.hooks.find(v => v.name === 'runUserEvents').func(currentFrameEvents);
+  this.tickCount += 1;
+  this.captureThumbnailImage();
```

Comment:
I'm a little worried about `tickCount` here.  (@joshlory opinion?) I think `onHandleEvents` only gets called when there's a reason to call back into student code, so we might find ourselves taking screenshots a lot more often in a complex project than we do in a simple one... or never, in a project that has no events after a certain frame?  Maybe that's desirable though.

I'd consider leaving your `captureThumbnailImage()` call here, but basing your capture interval on timing (`Date.now()` or `performance.now()`) instead of a tickCount that may not advance at a consistent rate.

We could also have dance-party pass the [`p5.frameCount`](https://p5js.org/reference/#/p5/frameCount) into this callback.

Reply:
okay, cool. it sounds like the quickest path forward would be to always call `captureThumbnailImage()`, then use a timestamp to determine whether we should take another thumbnail snapshot, storing the `lastThumbnailCapturedTimestamp`. 

i'm thinking capture a snapshot every 5 seconds, unless we have preferred timing already figured out.

Reply:
Nothing figured out AFAIK. 5 seconds sounds like a good starting point - I'd expect anything publish-worthy to have run for at least 5 seconds.